From 46f60d9347b887ca737b1e5325c339fcd6338c3a Mon Sep 17 00:00:00 2001
From: tstegema <tstegema@90b65887-3827-0410-9a23-83215b262276>
Date: Thu, 27 Jun 2013 20:02:38 +0000
Subject: [PATCH 0344/1000] appControl.idl V1.0.0.0" struct application is
 deprecated; retained for compatibility with old MCC applicationStart.cmdLine
 changed from String to List of Strings (more than one process per application
 allowed)

for CMD_START used applicationStart
for CMD_SWITCH, CMD_KILL, CMD_SHOW use applicationId

git-svn-id: svn://localhost/SelfServiceCommon/trunk@515 90b65887-3827-0410-9a23-83215b262276
---
 .../Massai/idl/Massai/appcontrol.idl          | 35 ++++++++++---------
 1 file changed, 19 insertions(+), 16 deletions(-)

diff --git a/SelfServiceCommon/Massai/idl/Massai/appcontrol.idl b/SelfServiceCommon/Massai/idl/Massai/appcontrol.idl
index 0bc02325..9bc63895 100644
--- a/SelfServiceCommon/Massai/idl/Massai/appcontrol.idl
+++ b/SelfServiceCommon/Massai/idl/Massai/appcontrol.idl
@@ -1,15 +1,9 @@
-//---------------------------------------------------------------------------
-//
-//  file:     appControl.idl
-//
-//  purpose:  basic definition for the massai application control service
-//            (internal)
-//
-//  date:     25.06.01
-//
-//  author:   (CMe/AGe)
-//
-//---------------------------------------------------------------------------
+// basic definition for the massai application control service  (internal)
+// M_INTERFACE_VERSION="1.0.0.0" 
+// History:
+// V1.0.0.0
+// application is deprecated; retained for compatibility with old MCC
+// applicationStart.cmdLine changed from String to List of Strings (more than one process per application allowed)
 
 #ifndef APPCONTROL_IDL
 #define APPCONTROL_IDL
@@ -29,6 +23,10 @@
 
 module appControl
 {
+    // list of strings
+    typedef sequence<string> strings;
+
+
     /** Definition of screen resolutions. */
 
     enum resolutions
@@ -40,14 +38,19 @@ module appControl
       RES_DEFAULT
     };
 
+    // deprecated for CMD_START used applicationStart, for CMD_SWITCH, CMD_KILL, CMD_SHOW use applicationId */
+    struct application
+    {
+      string appName;     /**< The application name as internally handeled by the ApplicationManager*/
+      string cmdLine;     /**< ignored; */ 
+    };
+
     /** Defines the application data that is passed by the ApplicationManager to this device
-     *  @note The cmdLine must be given as EXECUTABLE.EXE params: param1 param2 ...
      */
 
-    struct application
+    struct applicationId
     {
       string appName;     /**< The application name as internally handeled by the ApplicationManager*/
-      string cmdLine;     /**< The commandline for the application to start within the newly created dektop */
     };
 
     /** Defines the application data that is passed by the ApplicationManager to this device
@@ -57,7 +60,7 @@ module appControl
     struct applicationStart
     {
       string appName;     /**< The application name as internally handeled by the ApplicationManager*/
-      string cmdLine;     /**< The commandline for the application to start within the newly created dektop */
+      strings cmdLine;     /**< The commandline for the application to start within the newly created dektop */
       long switchTimeout; /**< timeout in ms; wait time before switching back to main, when process is started */
     };
 
-- 
2.41.0.windows.1

